packages/experimental/MenuButton/src/MenuButton.tsx

```diff
@@ -0,0 +1,71 @@
+/** @jsx withSlots */
+import React, { useRef, useState, useCallback } from 'react';
+import { Button } from '@fluentui-react-native/experimental-button';
+import { compose, UseSlots, withSlots } from '@fluentui-react-native/framework';
+import { Icon } from '@fluentui-react-native/icon';
+import { stylingSettings } from './MenuButton.styling';
+import ChevronIconSvg from './chevron.svg';
+import { renderContextualMenu } from './renderContextualMenu';
+
+import { menuButtonName, MenuButtonProps, MenuButtonType } from './MenuButton.types';
+
+export const MenuButton = compose<MenuButtonType>({
+  displayName: menuButtonName,
+  ...stylingSettings,
+  slots: {
+    root: React.Fragment,
+    chevronIcon: Icon,
+  },
+  slotProps: {
+    root: {},
+    chevronIcon: {
+      svgSource: {
+        src: ChevronIconSvg,
+      },
+      width: 12,
+      height: 16,
+    },
+  },
+  render: (props: MenuButtonProps, useSlots: UseSlots<MenuButtonType>) => {
+    const { menuItems, content, icon, disabled, onItemClick, contextualMenu, primary } = props;
+
+    const stdBtnRef = useRef(null);
+    const [showContextualMenu, setShowContextualMenu] = useState(false);
+    const onDismiss = useCallback(() => {
+      setShowContextualMenu(false);
+    }, [setShowContextualMenu]);
+    const toggleShowContextualMenu = useCallback(() => {
+      setShowContextualMenu(!showContextualMenu);
+    }, [showContextualMenu, setShowContextualMenu]);
+
+    const buttonProps = {
+      content,
+      disabled,
+      primary,
+      icon,
+      componentRef: stdBtnRef,
+      onClick: toggleShowContextualMenu,
+    };
+
+    const contextualMenuProps = {
+      onItemClick,
+      target: stdBtnRef,
+      onDismiss,
+      setShowMenu: toggleShowContextualMenu,
+      ...contextualMenu,
+    };
+
+    const Slots = useSlots(props);
+
+    return () => {
+      return (
+        <Slots.root>
+          <Button {...buttonProps}>
+            <Slots.chevronIcon />
+          </Button>
+          {showContextualMenu && renderContextualMenu(contextualMenuProps, menuItems)}
```

Comment:
I just realized, why isn't the contextual menu just a slot?

Reply:
It didn't work with slots. The properties inside child components were undefined. To make it work I had to add property slotProps, but in this case I couldn't customize the props.

Reply:
undefined child components sounds like a bug we ran into when we forgot to add the JSX pragma "withSlots" at the top of the file. Could you elaborate on the bug you got trying to use this as a slot? I. suspect trying to use slots and fragments are conflicting, like you mentioned they were

Reply:
I didn't use fragments and slots in this case, because it even wasn't bundling. Some slots are working, for the icon for example, because I use there predefined slotProps

Reply:
I would prefer to make decisions on how to structure our components on whether it makes sense technically to do so, not whether it builds or bundles. We should be able to work through the building issues you mentioned if it makes sense to use slots.

Looking into Slots, it seems like the benefit it brings is that it filters out extra entries in the DOM created by child components that make up the larger component. This reduces overhead on the DOM when rendering.

It sounds like something that wouldn't be as useful for the Button but could save some entries in the context menu? Testing this somehow would be nice, since I'm mostly guessing here. I'll poke around with the render tree and see what results.

This is a pretty useful diagram (it's from the old framework, but the structure should be similar to the new one):
https://github.com/microsoft/fluentui-react-native/blob/4687ff096fd25b07e0a4e782616b6b2ec0fa2d9f/packages/deprecated/foundation-composable/docs/Composable.png

Reply:
The problem with bundling wasn't about using slots. We can't use in one file 'jsx' and 'use slots' pragmas. The promlem of slots in this case that customized properties were undefined.
What about extra entries in the DOM, I'm not sure. I checked the render tree for the ContextualMenu with slots and without, it looks the same.
Thank you for the diagram :)